src/java/main/org/apache/zookeeper/server/ZooKeeperServer.java

```diff
@@ -303,15 +305,38 @@ public void loadData() throws IOException, InterruptedException {
 
     public void takeSnapshot(){
```

Comment:
If we are not expect to take snapshot on same time, the most easy way is to use `AtomicBoolean ` protect the `takeSnapshot` from all the code path, which may block the `SyncRequestProcessor` a while if a manual task is ongoing. My current code is assuming the background `SyncRequestProcessor` have higher priority.